### PR TITLE
FOGL-7260 - CentOS Stream9 fixes in core & their system tests

### PR DESCRIPTION
**Pull request by ashish-jabble**

### CODE REVIEW

File: python/fledge/common/utils.py [hunk 1]
```diff
@@ -85,3 +85,42 @@ def decorator(Class):
...
+    """ General information to identifying the operating system """
+    import ast
+    import re
+    os_details = {}
+    try:
+        filename = '/etc/os-release'
+        f = open(filename)
+    except FileNotFoundError:
+        filename = '/usr/lib/os-release'
+        f = open(filename)
```
Comment (praveen-garg) [inline python/fledge/common/utils.py:100]: use with context or close after reading?

File: python/fledge/common/utils.py [hunk 2]
```diff
@@ -85,3 +85,42 @@ def decorator(Class):
...
+
+    for line_number, line in enumerate(f, start=1):
+        line = line.rstrip()
+        if not line or line.startswith('#'):
+            continue
+        m = re.match(r'([A-Z][A-Z_0-9]+)=(.*)', line)
```
Comment (praveen-garg) [inline python/fledge/common/utils.py:106]: add comment for regex explaination

File: python/fledge/common/utils.py [hunk 3]
```diff
@@ -85,3 +85,42 @@ def decorator(Class):
...
+            name, val = m.groups()
+            if val and val[0] in '"\'':
+                val = ast.literal_eval(val)
+            os_details.update({name: val})
+    return os_details
+
+
+def is_debian():
+    """
+        To check if the Operating system is of Debian type or Not
```
Comment (praveen-garg) [inline python/fledge/common/utils.py:117]: of Debian type or Not
Debian based or Not

Actually we are checking if Redhat family or not

**Review by praveen-garg — COMMENTED** [review]
minor comments otherwise lgtm!

File: python/fledge/common/utils.py (hunk 3 above)
Comment (praveen-garg) [reply python/fledge/common/utils.py:111]: is_redhat_based/is_redhat_family will keep it more cleaner along with existing changed for yum vs apt.

e.g. 
`pkg_mgt = 'yum' if is_redhat_based() else 'apt'`

File: python/fledge/services/core/api/update.py [hunk 4]
```diff
@@ -126,10 +125,10 @@ async def get_updates(request: web.Request) -> web.Response:
...
          curl -sX GET http://localhost:8081/fledge/update |jq
     """
-    _platform = platform.platform()
-    update_cmd = "sudo apt update"
-    upgradable_pkgs_check_cmd = "apt list --upgradable | grep \^fledge"
-    if "centos" in _platform or "redhat" in _platform:
+    if utils.is_debian():
+        update_cmd = "sudo apt update"
+        upgradable_pkgs_check_cmd = "apt list --upgradable | grep \^fledge"
+    else:
```
Comment (praveen-garg) [inline python/fledge/services/core/api/update.py:131]: that way we can also keep here default as is; and  only `"centos" in _platform or "redhat" in _platform` with if is replaced.

**Review by praveen-garg — APPROVED** [review]

**Review by MarkRiddoch — APPROVED** [review]

**Pull request merged**